src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/RemoveAzureContainerRegistryReplication.cs

```diff
@@ -0,0 +1,75 @@
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsCommon.Remove, ContainerRegistryReplicationNoun, SupportsShouldProcess = true)]
+    public class RemoveAzureContainerRegistryReplication : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Replication Name. Default to the location name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ReplicationNameAlias)]
+        public string Name { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 2, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string RegistryName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ReplicationObjectParameterSet, ValueFromPipeline = true, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
+        public PSContainerRegistryReplication Replicatoin { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The container registry replication resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ResourceIdAlias)]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, "Delete the replication from the container registry"))
+            {
+                if(string.Equals(ParameterSetName, ReplicationObjectParameterSet))
+                {
+                    ResourceId = Replicatoin.Id;
+                }
+                if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
+                {
+                    string resourceGroup, registryName, childResourceName;
+                    if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName)
+                        || string.IsNullOrEmpty(childResourceName))
+                    {
+                        WriteInvalidResourceIdError(InvalidReplicationResourceIdErrorMessage);
+                        return;
+                    }
+
+                    ResourceGroupName = resourceGroup;
+                    RegistryName = registryName;
+                    Name = childResourceName;
+                }
+
+                if (ShouldProcess(Name, "Remove Container Registry Replication"))
```

Comment:
@Jinhuafei this `ShouldProcess` block is unnecessary since it was already called on line 47

I think it would be best if you remove the `ShouldProcess` call on line 47 and have the actual client call (`RegistryClient.DeleteReplication(...)`) be wrapped in the only `ShouldProcess` block, as it is now.

Reply:
good catch.